### PR TITLE
feat: add task minAvailable design

### PR DESCRIPTION
**Pull request by shinytang6**
Signed-off-by: shinytang6 <1074461480@qq.com>

### CODE REVIEW

**Comment by shinytang6** [comment]
#1131 (the first step in my design) will cause some compatibility issues indeed, but l think it makes sense in some way.
WDYT? @Thor-wl @william-wang

**Comment by shinytang6** [comment]
@Thor-wl @k82cn @william-wang PTAL.

File: docs/design/task-minavailable.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        ...
+       
+        // MinTaskMember defines the minimal number of pods to run each task in the pod group;
+        // if there's not enough resources to start each task, the scheduler
+        // will not start anyone.
+        MinTaskMember map[string]int32
+   }
+   ```
+3. Add a new judgment logic to the current gang scheduling. The current logic is if sumof(valid tasks) >= job.minavailable, we take this job as valid(can be scheduled). We need to add a logic before that. The logic is that if the `minAvailable` field of the task is set, the task under current job must meet the conditions of (valid pod of the task) >= job.task.minAvailable, then we can take the job as valid.
+4. Modify the judgment of job status. The current logic is that if the sumof(successful pods) >= job.minAvailable, then we can take the status of the job as `Completed`. This change may need another field in `JobStatus` to record the status of the pods under each task, currently `JobStatus` only records the number of pods in different states which we cannot distinguish which task pod belongs to.
```
Comment (Thor-wl) [inline docs/design/task-minavailable.md:48]: an optional point: adjust CLI to the feature and support getting pods statistics of task level.

**Comment by Thor-wl** [comment]
Has this feature been implemented? @shinytang6

**Comment by shinytang6** [comment]
> Has this feature been implemented? @shinytang6

[PR 1179](https://github.com/volcano-sh/volcano/pull/1193) implemented this feature, but it's still under test, not fully ready.

**Comment by Thor-wl** [comment]
> > Has this feature been implemented? @shinytang6
> 
> [PR 1179](https://github.com/volcano-sh/volcano/pull/1193) implemented this feature, but it's still under test, not fully ready.

OK, please finish it as soon as possible for it is an important feature in v1.2

**Comment by Thor-wl** [comment]
This feature is very important and we plan to merge it in v1.3.0

**Comment by shinytang6** [comment]
Please also help review that
/cc @zen-xu

**Review by zen-xu — APPROVED** [review]

File: docs/design/task-minavailable.md (hunk 1 above)
Comment (Thor-wl) [inline docs/design/task-minavailable.md:48]: Another concern: we should consider the releationship between job.minAvailable and the sum of minAvailable at task levels. If job.minAvailable is less, the judgement can be ignored.

**Review by Thor-wl — APPROVED** [review]

**Pull request merged**